liota/core/package_manager.py

```diff
@@ -30,4 +30,1104 @@
+            if not isinstance(dependencies, list):
+                log.error("Mal-formatted list of dependencies in module %s" \
+                        % module_loaded.__name__)
+                return None
+
+            if len(dependencies) > 0:
+                log.info("Package %s depends on: %s" \
+                        % (file_name, " ".join(dependencies)))
+                if not isinstance(check_stack, list):
+                    check_stack = []
+                check_stack.append(file_name)
+                for dependency in dependencies:
+                    if dependency in check_stack:
+                        log.error("%s is not loaded, because %s depends on it" \
+                                % (file_name, dependency))
+                        check_stack.pop()
+                        return None
+                    if not dependency in self._packages_loaded:
+                        self._package_load(dependency, check_stack=check_stack)
+                    if not dependency in self._packages_loaded:
+                        log.error("%s is not loaded, because %s failed to load"\
+                                % (file_name, dependency))
+                        check_stack.pop()
+                        return None
+                    
+                    # Add dependent record
+                    dep_record = self._packages_loaded[dependency]
+                    assert(isinstance(dep_record, PackageRecord))
+                    dep_record.add_dependent(file_name)
+                check_stack.pop()
+                log.debug("Dependency check of package %s is complete" \
+                        % file_name)
+
+        # Get package class from module and instantiate it
+        if not hasattr(module_loaded, "PackageClass"):
+            log.error("Invalid package: %s" % module_name)
+            return None
+
+        klass = getattr(module_loaded, "PackageClass")
+        package_record = PackageRecord(file_name)
+        if not package_record.set_instance(klass()):
+            log.error("Unexpected failure initializing package")
+            return None
+        try: # Run created instance
+            package_record.get_instance().run(
+                    self._resource_registry.get_package_registry(file_name)
+                )
+        except Exception as er:
+            log.error("Exception in initialization: %s" % str(er))
+            return None
+        package_record.set_sha1(sha1)
+        package_record.set_ext(file_ext)
+        package_record.set_dependencies(dependencies)
+        self._packages_loaded[file_name] = package_record
+
+        log.info("Package class from module %s is initialized" \
+                % module_loaded.__name__)
+        return package_record
+
+    #-----------------------------------------------------------------------
+    # This method is called to unload package using its file_name (no ext).
+
+    def _package_unload(self, file_name, track_list=None):
```

Comment:
Need to define the parameters like track_list. I didn't understand its use until i reached package_reload.